In [1]:
import pandas as pd 
import scipy
import matplotlib.pyplot as plt
from sklearn.preprocessing import LabelEncoder
 

In [6]:
train_df = pd.read_csv("./Dataset/train.csv")
test_df = pd.read_csv("./Dataset/test.csv")

In [7]:
train_df_c = train_df.copy()
test_df_c = train_df.copy()

In [4]:
train_df_c.head()

Student ID   Age Gender   Home Region  Home City  \
0  4f14c50d-162e-4a15-9cf0-ec129c33bcf0  37.0    ذكر  منطقة الرياض     الرياض   
1  0599d409-876b-41a5-af05-749ef0e77d32  21.0    ذكر    منطقة عسير  خميس مشيط   
2  38a11c0e-4afc-4261-9c64-e94cc0a272fb  24.0    ذكر  منطقة الرياض     الرياض   
3  1693e85b-f80e-40ce-846f-395ddcece6d3  23.0    ذكر  منطقة الرياض     الرياض   
4  98a0e8d0-5f80-4634-afd8-322aa0902863  23.0    ذكر  منطقة الرياض     الرياض   

                             Program ID Program Main Category Code  \
0  453686d8-4023-4506-b2df-fac8b059ac26                       PCRF   
1  cc8e4e42-65d5-4fa1-82f9-6c6c2d508b60                       APMR   
2  e006900d-05a9-4c2b-a36f-0ffb9fce44cd                       APMR   
3  2ec15f6b-233b-428a-b9f5-e40bc8d14cf9                       TOSL   
4  d32da0e9-1aed-48c3-992d-a22f9ccc741e                       CAUF   

  Program Sub Category Code Technology Type Program Skill Level  ...  \
0                      PCRF             NaN                 NaN  ...   
1                      SWPS             NaN               متوسط  ...   
2                       NaN             NaN               متوسط  ...   
3                      TOSL             NaN                 NaN  ...   
4                      SWPS         تقليدية               متوسط  ...   

  Completed Degree Level of Education                Education Speaciality  \
0              نعم        البكالوريوس                     هندسة حاسب الالي   
1              نعم        البكالوريوس           الإذاعة والتلفزيون والفيلم   
2              نعم        البكالوريوس               Information Technology   
3              نعم        البكالوريوس  حوسبة تطبيقية - (مسار شبكات الحاسب)   
4               لا        البكالوريوس              نظم المعلومات الحاسوبية   

                          College University Degree Score  \
0                             NaN                    2.44   
1        الفنون والعلوم الإنسانية                    5.00   
2                             NaN                    3.50   
3                             NaN                    3.55   
4  تكنولوجيا الاتصالات والمعلومات                    4.00   

  University Degree Score System Employment Status Job Type  Still Working  Y  
0                            4.0          غير موظف      NaN            NaN  0  
1                            5.0              طالب      NaN            NaN  0  
2                            5.0              موظف      NaN            NaN  0  
3                            5.0              خريج      NaN            NaN  0  
4                            5.0               NaN      NaN            NaN  0  

[5 rows x 24 columns]

# Invistaging if there's any duplicates 

In [19]:
def check_dublicate(train,test):
    """
    check wither theres a 'leak' in the test set from the train set

    Args:
        train (_type_): dataframe
        test (_type_): dataframe
        
    Return: 
        bool: True if there are leaked records, False otherwise.
        list: A list of leaked records (if any).
        float: The percentage of leaked records (if any).
    """
    hash_train = set(train)
    hash_test = set(test)
    
    leaks = list(hash_train.intersection(hash_test))
    has_leak = len(leaks) > 0
    print(len(leaks))
    
    # Percentage
    num_leak = len(leaks)
    total_records_train = len(train)
    leak_percentage = (num_leak / total_records_train) * 100
    
    return has_leak, leaks , leak_percentage  
    
has_leak, _ ,leak_percentage = check_dublicate(train_df_c['Student ID'],test_df_c['Student ID'])

print(has_leak,leak_percentage)

5196
True 79.35247403787416


In [ ]:
train_df_c.info

In [12]:
test_df_c[test_df_c['Student ID'] == '1ad6b961-5e47-4985-ba40-547f9ecef363']

Student ID   Age Gender   Home Region  \
482   1ad6b961-5e47-4985-ba40-547f9ecef363  25.0    ذكر  منطقة الرياض   
3178  1ad6b961-5e47-4985-ba40-547f9ecef363  25.0    ذكر  منطقة الرياض   

     Home City                            Program ID  \
482     الرياض  2ef39820-72d1-4d8f-be97-2d71f3737685   
3178    الرياض  55a8f0c7-7fdc-4574-9828-44e5979f7c78   

     Program Main Category Code Program Sub Category Code Technology Type  \
482                        PCRF                      PCRF         تقليدية   
3178                       PCRF                      PCRF         تقليدية   

     Program Skill Level  ... Completed Degree Level of Education  \
482                متقدم  ...              نعم        البكالوريوس   
3178               متقدم  ...              نعم        البكالوريوس   

         Education Speaciality  College University Degree Score  \
482   حاسب الي - نظم المعلومات      NaN                    4.45   
3178  حاسب الي - نظم المعلومات      NaN                    4.45   

     University Degree Score System Employment Status Job Type  Still Working  \
482                             5.0          غير موظف      NaN            NaN   
3178                            5.0          غير موظف      NaN            NaN   

      Y  
482   0  
3178  0  

[2 rows x 24 columns]

In [13]:
train_df[train_df['Student ID'] == '1ad6b961-5e47-4985-ba40-547f9ecef363']

Student ID   Age Gender   Home Region  \
482   1ad6b961-5e47-4985-ba40-547f9ecef363  25.0    ذكر  منطقة الرياض   
3178  1ad6b961-5e47-4985-ba40-547f9ecef363  25.0    ذكر  منطقة الرياض   

     Home City                            Program ID  \
482     الرياض  2ef39820-72d1-4d8f-be97-2d71f3737685   
3178    الرياض  55a8f0c7-7fdc-4574-9828-44e5979f7c78   

     Program Main Category Code Program Sub Category Code Technology Type  \
482                        PCRF                      PCRF         تقليدية   
3178                       PCRF                      PCRF         تقليدية   

     Program Skill Level  ... Completed Degree Level of Education  \
482                متقدم  ...              نعم        البكالوريوس   
3178               متقدم  ...              نعم        البكالوريوس   

         Education Speaciality  College University Degree Score  \
482   حاسب الي - نظم المعلومات      NaN                    4.45   
3178  حاسب الي - نظم المعلومات      NaN                    4.45   

     University Degree Score System Employment Status Job Type  Still Working  \
482                             5.0          غير موظف      NaN            NaN   
3178                            5.0          غير موظف      NaN            NaN   

      Y  
482   0  
3178  0  

[2 rows x 24 columns]

Insights
- yes there's a leak and it's not normal the leak is 5196 record out of 6548, 79%
- it might be ***usefull*** to copy the solution of (get inspired by it) https://www.kaggle.com/competitions/arabic-poem-classification/discussion/483099